README.md

```diff
@@ -1,68 +1,256 @@
+* The "Content type" field should be `application/json`.
+* The "secret" field should match bot `config::github_webhook_secret`.
+* The bot needs to receive `Pull request`, `Pull request review`, and `Status` events.
+
+
+### Bot configuration
+
+The bot loads its configuration from `./config.json`. You may use
+`config-example.json` to bootstrap your configuration.
+
+All configuration fields are required (XXX: Why is there a "default" column in the table below then? Rename "default" to example and adjust accordingly?).
+
+*Field* | *Description* | *default*
+--- | --- | ---
+*github_username* | The bot uses this GitHub user account for all GitHub communications, including target branch updates. This user needs to have write access to the repository.| -
+*github_token* | An authentication token generated for the associated `config::github_username`. | -
+*github_webhook_path* | GitHub webhook URL path. | -
+*github_webhook_secret* | A random secret string to be used here and in the GitHub webhook configuration. | -
+*port* | The bot listens for GitHub requests on this TCP port. | 7777
+*repo* | The name of the GitHub repository that the bot should serve. | -
+*owner* | The owner (a person or organization) of the GitHub repository. | -
+*dry_run*| A boolean option to enable read-only, no-modifications mode where the bot logs pull requests selected for merging but skips further merging steps.| false
+*skip_merge*| A boolean option to enable no-final-modifications mode where the bot performs all the merging steps up to (and not including) the target branch update. Eligible PRs are merged into and tested on the staging branch but are never merged into their target branches. | false
+*staging_branch* | The name of the staging branch. | auto
+*necessary_approvals* | The minimal number of "core" developers required for a PR to become eligible for merging. PRs with fewer votes are not merged, regardless of their age. | 2
+*voting_delay_min*| The minimum merging age of a PR. Younger PRs are not merged, regardless of the number of votes. PR age is measured in days from the PR creation time. | 2
+*sufficient_approvals* | The minimal number of core developers required for a PR to be merged fast (i.e., without waiting for `config::voting_delay_max`) | 2
+*voting_delay_max* | The minimum merging age of a PR that has fewer than `config::sufficient_approvals` votes. PR age is measured in days from the PR creation time. | 10
+
+
+## Caveats
+
+### Merging eligibility may change while merging
+
+The bot checks merge eligibility conditions when starting to process a
+given pull request. It is possible that a checked condition changes
+during or after that initial check. There is no guarantee that the bot
+will notice such "late" changes. Similar race conditions exist for
+manual PR merging, of course.
+
+TODO: Should the bot recheck the conditions again after all staging
+branch tests are over?
```

Comment:
I don't think it should. I master changes, the FF merge fails and we will notice that. If branch changes, we should not care about that: we merge one of its the previous revisions.

Reply:
Done at 3a063d4.